integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/context/APIManagerWithMgwServerInstance.java

```diff
@@ -0,0 +1,90 @@
+import org.wso2am.micro.gw.tests.util.TestConstant;
+import org.wso2am.micro.gw.tests.util.Utils;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * APIM with Mgw server instance class.
+ */
+public class APIManagerWithMgwServerInstance extends MgwServerImpl {
+
+    /**
+     * initialize a docker environment using docker compose.
+     *
+     * @throws IOException          if something goes wrong while adding the mock backend to the docker-compose.yaml
+     * @throws MicroGWTestException if something goes wrong while copying server configs
+     */
+    public APIManagerWithMgwServerInstance() throws IOException, MicroGWTestException {
+        this(null, false);
+    }
+
+    /**
+     * initialize a docker environment using docker compose.
+     *
+     * @param confPath external conf.toml path
+     * @throws IOException          if something goes wrong while adding the mock backend to the docker-compose.yaml
+     * @throws MicroGWTestException if something goes wrong while copying server configs
+     */
+    public APIManagerWithMgwServerInstance(String confPath) throws IOException, MicroGWTestException {
+        this(confPath, false);
+    }
+
+    /**
+     * initialize a docker environment using docker compose.
+     *
+     * @param confPath   - external conf.toml path
+     * @param tlsEnabled - if the backend needs to have the tls enabled server additionally
+     * @throws IOException          if something goes wrong while adding the mock backend to the docker-compose.yaml
+     * @throws MicroGWTestException if something goes wrong while copying server configs
+     */
+    public APIManagerWithMgwServerInstance(String confPath, boolean tlsEnabled) throws MicroGWTestException,
+                                                                                       IOException {
+        createTmpMgwSetup();
+        if (!StringUtils.isEmpty(confPath)) {
+            Utils.copyFile(confPath, mgwTmpServerPath + File.separator + "resources" + File.separator +
+                    "conf" + File.separator + "config.toml");
+        }
+        String dockerComposePath =
+                mgwTmpServerPath + File.separator + "docker-with-apim" + File.separator + "docker-compose.yaml";
+        // add mock backend service to the docker-compose.yaml file
+        MockBackendServer.addMockBackendServiceToDockerCompose(dockerComposePath, tlsEnabled);
+
+        environment = new DockerComposeContainer(new File(dockerComposePath)).withLocalCompose(true).waitingFor(
+                TestConstant.APIM_SERVICE_NAME_IN_DOCKER_COMPOSE,
+                Wait.forLogMessage("\\/localhost:9443\\/carbon\\/", 1));
+    }
+
+    /**
+     * This will create a separate mgw setup in the target/server-tmp directory to execute the tests.
+     *
+     * @throws MicroGWTestException if something goes wring while copying the config file
+     */
+    private void createTmpMgwSetup() throws MicroGWTestException {
```

Comment:
Shall we move this to the previous location because this is not APIM specific?

Reply:
This function is not removed from the previous location (MgwServerInstance.java). Just the method signature is changed there.

Reply:
This function is not removed from the previous location (MgwServerInstance.java). Just the method signature is changed there.

Reply:
Sorry, I meant the content below it. Edited my comment above to suit the scenario.

Reply:
The content where it copies the extracted mgw to a temp location.

Reply:
+1